### PR TITLE
net: Hardcoded seeds update for 0.18

### PR DESCRIPTION
**Pull request by laanwj**
I've followed the steps in https://github.com/bitcoin/bitcoin/blob/master/contrib/seeds/README.md but it seems all the hardcoded onion peers except for two are being erased, for some reason. 

~~**Do not merge as-is**~~ (might be ready after including results from multiple seeds)

### CODE REVIEW

**Comment by laanwj** [comment]
Added some debugging of the funnel
```
Initial: IPv4 407890, IPv6 56492, Onion 715
Skip entries with valid address: IPv4 407890, IPv6 56492, Onion 715
Skip entries from suspicious hosts: IPv4 407889, IPv6 56492, Onion 715
Enforce minimal number of blocks: IPv4 104845, IPv6 48821, Onion 617
Require service bit 1: IPv4 104650, IPv6 48759, Onion 607
Require at least 50% 30-day uptime: IPv4 4961, IPv6 966, Onion 19
Require a known and recent user agent: IPv4 3408, IPv6 628, Onion 2
Filter out hosts with multiple bitcoin ports: IPv4 3402, IPv6 624, Onion 2
ERR: Could not resolve ASN for "91.195.42.134"
ERR: Could not resolve ASN for "150.187.36.233"
Look up ASNs and limit results, both per ASN and globally: IPv4 512, IPv6 624, Onion 2
```

These are the onion-killers:
- Require at least 50% 30-day uptime
- Require a known and recent user agent

**Comment by laanwj** [comment]
would be useful to have `seeds.txt.gz` files from the vintage point of other crawlers, to see if this drop in onion nodes and their reliability is an actual thing

@TheBlueMatt @luke-jr @cdecker @jonasschnelli @petertodd @Sjors @emzy

**Comment by Emzy** [comment]
Here is mine.
[seeds_emzy.txt.gz](https://github.com/bitcoin/bitcoin/files/3068703/seeds_emzy.txt.gz)

**Comment by laanwj** [comment]
thank you @Emzy!
however, your list doesn't have onion addresses at all

**Comment by Emzy** [comment]
New file with onion addresses. ("-o" option on)
[seeds_emzy.txt.gz](https://github.com/bitcoin/bitcoin/files/3068924/seeds_emzy.txt.gz)

**Comment by luke-jr** [comment]
Mine is here: https://luke.dashjr.org/programs/bitcoin/files/charts/data/seeds.txt (161 MB)

**Comment by petertodd** [comment]
[dnsseed.dump.gz](https://github.com/bitcoin/bitcoin/files/3069437/dnsseed.dump.gz)

**Comment by laanwj** [comment]
Thanks, combining all those, removing duplicates, and running them through `./makeseeds.py` ends up with: IPv4 512, IPv6 781, Onion 171

This is okay—the old list had 176 onion hosts, and 1092 IPv4+IPv6.

Pushed the new list.

**Comment by Sjors** [comment]
My seed found 587 onion nodes in the past couple of hours, 26 of which are marked "good". Should I just upload the list here, or give it a bit more time?

**Comment by Sjors** [comment]
Wouldn't this strip all 0.17 and 0.18 nodes (or did you manually tweak that)?
https://github.com/bitcoin/bitcoin/blob/8a9ffec0a257da659ba54c5073bfc820986ae9c1/contrib/seeds/makeseeds.py#L33

File: contrib/seeds/makeseeds.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 PATTERN_IPV4 = re.compile(r"^((\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3})):(\d+)$")
 PATTERN_IPV6 = re.compile(r"^\[([0-9a-z:]+)\]:(\d+)$")
 PATTERN_ONION = re.compile(r"^([abcdefghijklmnopqrstuvwxyz234567]{16}\.onion):(\d+)$")
-PATTERN_AGENT = re.compile(r"^(/Satoshi:0.14.(0|1|2|99)/|/Satoshi:0.15.(0|1|2|99)|/Satoshi:0.16.(0|1|2|99)/)$")
+PATTERN_AGENT = re.compile(r"^(/Satoshi:0.14.(0|1|2|3|99)/|/Satoshi:0.15.(0|1|2|99)|/Satoshi:0.16.(0|1|2|3|99)/|/Satoshi:0.17.(0|1|2|99)/|/Satoshi:0.18.99/)$")
```
Comment (Sjors) [inline contrib/seeds/makeseeds.py]: We should add an instruction to add new version here in /contrib/seeds/README.md.

**Review by Sjors — COMMENTED** [review]
Should we update the 4 year old `SUSPICIOUS_HOSTS` list?

Maybe use a scripted-diff for the `generate-seeds.py` command (which produces the hard to read `.h` file)?

**Comment by cdecker** [comment]
> would be useful to have seeds.txt.gz files from the vintage point of other crawlers, to see if this drop in onion nodes and their reliability is an actual thing

Since my seed uses a custom crawler and DNS implementation I can't provide the file in question, but I'm more than happy to check proposed nodes for uptime in my database :-)

**Comment by jonasschnelli** [comment]
I also don't run with `-o`
[dnsseed.dump.zip](https://github.com/bitcoin/bitcoin/files/3072098/dnsseed.dump.zip)

File: contrib/seeds/makeseeds.py (hunk 1 above)
Comment (laanwj) [reply contrib/seeds/makeseeds.py]: There is one:

> Be sure to update `PATTERN_AGENT` in `makeseeds.py` to include the current version,
and remove old versions as necessary (at a minimum when GetDesirableServiceFlags
changes its default return value, as those are the services which seeds are added
to addrman with).

**Comment by laanwj** [comment]
> Should we update the 4 year old SUSPICIOUS_HOSTS list?

Possibly. I think maintaining this as part of the repository at all is a bit questionable, I don't think bitcoin core development should focus on judging individual nodes. But if you have any suggestions, feel free to add them.

> Maybe use a scripted-diff for the generate-seeds.py command (which produces the hard to read .h file)?

I think this is a good idea but maybe do this for a future release, after updating the instructions how to do this?
For now, can verify the list by running the `python3 generate-seeds.py . > ../../src/chainparamsseeds.h` command yourself.

Apart from that, for review please focus on the `contrib/seeds/nodes_main.txt` list.

**Comment by practicalswift** [comment]
Concept ACK on removing `SUSPICIOUS_HOSTS`

Given the current number of nodes we should be robust against the presence of a few bad IP:s?

**Comment by laanwj** [comment]
> Concept ACK on removing SUSPICIOUS_HOSTS

Yes—I think this is a valid discussion but please move it to a new issue. Here we just want the things for rc4 (and this was a part of the release process that was forgotten) in so that we can continue with the release :-)

**Comment by jonatack** [comment]
Without the `-o` option:
[2019-04_dnsseed_0-18-0-rc3.dump.gz](https://github.com/jonatack/bitcoin-development/blob/master/2019-04_dnsseed_bitcoin_0-18-0-rc3.dump.gz)

File: contrib/seeds/makeseeds.py (hunk 1 above)
Comment (luke-jr) [inline contrib/seeds/makeseeds.py]: Should we perhaps exclude nodes vulnerable to CVE-2018-17144?

Comment (luke-jr) [reply contrib/seeds/makeseeds.py]: And why is this only including Core nodes in the first place?

Comment (Sjors) [reply contrib/seeds/makeseeds.py]: CVE-2018-17144: for their own protection?

**Comment by laanwj** [comment]
Closing this: doesn't seem to be an improvement over the current state, so would be better to skip it this time, and discuss how to handle this before next major release…

**Comment by Sjors** [comment]
@laanwj for 0.19:

[dnsseed.dump.tar.gz](https://github.com/bitcoin/bitcoin/files/3671913/dnsseed.dump.tar.gz)
[dnsseed-testnet.dump.tar.gz](https://github.com/bitcoin/bitcoin/files/3671914/dnsseed-testnet.dump.tar.gz)

**Comment by laanwj** [comment]
@Sjors thanks!

**Pull request closed without merging**